File: CS/App_Code/Product.cs
using System;
using System.Collections.Generic;

[Serializable]
public class Product : IEquatable<Product> {
    public int ProductID { get; set; }
    public string ProductName { get; set; }
    public int UnitsOnOrder { get; set; }
    public decimal UnitPrice { get; set; }

    public Product(int id, string name, int unitsOnOrder, decimal price) {
        ProductID = id;
        ProductName = name;
        UnitsOnOrder = unitsOnOrder;
        UnitPrice = price;
    }
    public Product() { }

    public bool Equals(Product other) {
        return this.ProductID.Equals(other.ProductID);
    }
}

File: VB/App_Code/Product.vb
Imports System
Imports System.Collections.Generic

<Serializable> _
Public Class Product
    Implements IEquatable(Of Product)

    Public Property ProductID() As Integer
    Public Property ProductName() As String
    Public Property UnitsOnOrder() As Integer
    Public Property UnitPrice() As Decimal

    Public Sub New(ByVal id As Integer, ByVal name As String, ByVal unitsOnOrder As Integer, ByVal price As Decimal)
        ProductID = id
        ProductName = name
        Me.UnitsOnOrder = unitsOnOrder
        UnitPrice = price
    End Sub
    Public Sub New()
    End Sub

    Public Overloads Function Equals(ByVal other As Product) As Boolean Implements IEquatable(Of Product).Equals
        Return Me.ProductID.Equals(other.ProductID)
    End Function
End Class